plugins/org.eclipse.xtend.core/src/org/eclipse/xtend/core/macro/ConstantExpressionsInterpreter.xtend

```diff
@@ -107,7 +107,7 @@ class ConstantExpressionsInterpreter extends AbstractConstantExpressionsInterpre
 			val section = importSectionLocator.getImportSection(expression.eResource as XtextResource)
 			if(section != null) {
 				for (imp : section.importDeclarations) {
-					if(imp.static) {
+					if(imp.static && imp.importedTypeName != null) {
```

Comment:
I'd rather store the importedTypeName in a local var since it's (kind of) expensive to compute from the node model.